[Bug]: Android build failures

### Platform

Android

### Plugin

share_plus, device_info_plus

### Version

8.0.0 for share_plus and 10.0.0 for device_info_plus

### Flutter SDK

3.13.4

### Steps to reproduce

Tried to just update plugins in one of projects I work on.
Project uses fresh dependencies and has Kotlin 1.8.20 and AGP 7.4.2, so it is not an issue with some old Android native dependencies.

`flutter clean` didn't help.

Tried to find something similar and found that `mobile_scanner` also has a similar issue https://github.com/juliansteenbakker/mobile_scanner/issues/729 and it is also related to Kotlin 1.9.x

Not sure about the solution, but I also found [this page](https://developer.android.com/build/kotlin-support) where it is stated that for Kotlin 1.9 project needs to have AGP 8.0.

Based on this fact I am thinking about retracting latest versions of plugins that have Kotlin 1.9.10 and would like to revert Kotlin to older version to not have such unexpected failures that can't be reproduced by building available example apps.

### Code Sample

_No response_

### Logs

```shell
ERROR:D8: com.android.tools.r8.kotlin.H

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:mergeExtDexDebug'.
> Could not resolve all files for configuration ':app:debugRuntimeClasspath'.
   > Failed to transform kotlin-stdlib-1.9.10.jar (org.jetbrains.kotlin:kotlin-stdlib:1.9.10) to match attributes {artifactType=android-dex, asm-transformed-variant=NONE, dexing-enable-desugaring=true, dexing-enable-jacoco-instrumentation=false, dexing-is-debuggable=true, dexing-min-sdk=21, org.gradle.category=library, org.gradle.libraryelements=jar, org.gradle.status=release, org.gradle.usage=java-runtime}.
      > Execution failed for DexingWithClasspathTransform: /Users/vb/.gradle/caches/transforms-3/28d185a8757e9cd8ecae045e483e8213/transformed/jetified-kotlin-stdlib-1.9.10.jar.
         > Error while dexing.
```


### Flutter Doctor

```shell
[✓] Flutter (Channel stable, 3.13.4, on macOS 13.5.2 22G91 darwin-arm64, locale en-UA)
    • Flutter version 3.13.4 on channel stable at /Users/vb/fvm/versions/3.13.4
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 367f9ea16b (4 weeks ago), 2023-09-12 23:27:53 -0500
    • Engine revision 9064459a8b
    • Dart version 3.1.2
    • DevTools version 2.25.0

[!] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    • Android SDK at /Users/vb/Library/Android/sdk
    ✗ cmdline-tools component is missing
      Run `path/to/sdkmanager --install "cmdline-tools;latest"`
      See https://developer.android.com/studio/command-line for more details.
    ✗ Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/docs/get-started/install/macos#android-setup for more details.

[✓] Xcode - develop for iOS and macOS (Xcode 15.0)
    • Xcode at /Applications/Xcode-15.0.0.app/Contents/Developer
    • Build 15A240d
    • CocoaPods version 1.13.0

[✓] IntelliJ IDEA Ultimate Edition (version 2023.2.2)
    • IntelliJ at /Users/vb/Applications/IntelliJ IDEA Ultimate.app
    • Flutter plugin version 75.1.4
    • Dart plugin version 232.9559.10
```


### Checklist before submitting a bug

- [X] I searched issues in this repository and couldn't find such bug/problem
- [X] I Google'd a solution and I couldn't find it
- [X] I searched on StackOverflow for a solution and I couldn't find it
- [X] I read the README.md file of the plugin
- [X] I'm using the latest version of the plugin
- [X] All dependencies are up to date with `flutter pub upgrade`
- [X] I did a `flutter clean`
- [X] I tried running the example project

--- a/packages/share_plus/share_plus/android/build.gradle
+++ b/packages/share_plus/share_plus/android/build.gradle
@@ -2,7 +2,7 @@ group 'dev.fluttercommunity.plus.share'
 version '1.0-SNAPSHOT'
 
 buildscript {
-    ext.kotlin_version = '1.9.10'
+    ext.kotlin_version = '1.7.22'
     repositories {
         google()
         mavenCentral()

